cmd/generate/config/rules/sentry.go

```diff
@@ -22,3 +24,63 @@ func SentryAccessToken() *config.Rule {
 	tps := utils.GenerateSampleSecrets("sentry", secrets.NewSecret(utils.Hex("64")))
 	return utils.Validate(r, tps, nil)
 }
+
+func SentryOrgToken() *config.Rule {
+	// define rule
+	r := config.Rule{
+		RuleID:      "sentry-org-token",
+		Description: "Found a Sentry.io Organization Token, risking unauthorized access to error tracking services and sensitive application data.",
+		Regex:       regexp.MustCompile(`\bsntrys_eyJpYXQiO[a-zA-Z0-9+/]{10,200}(?:LCJyZWdpb25fdXJs|InJlZ2lvbl91cmwi|cmVnaW9uX3VybCI6)[a-zA-Z0-9+/]{10,200}={0,2}_[a-zA-Z0-9+/]{43}\b`),
+		// base64-encoded json fragments: eyJpYXQiO = `{"iat":`,
+		// depending on alignment: LCJyZWdpb25fdXJs = `,"region_url`, InJlZ2lvbl91cmwi = `"region_url"`, cmVnaW9uX3VybCI6 = `region_url":`
+
+		Entropy:  3.5,
+		Keywords: []string{"sntrys_eyJpYXQiO"},
+	}
+
+	// validate
+	tps := utils.GenerateSampleSecrets("sentry",
+		`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`, // gitleaks:allow
+	)
+	tps = append(tps,
+		` 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`, // gitleaks:allow
+	)
+
+	encodedJson := base64.StdEncoding.EncodeToString([]byte(secrets.NewSecret(
+		`\{"iat":[\d\.]{3,6},"url":"https://\w{10,20}","region_url":"https://\w{20,30}","org":"\w{5,10}"\}`)))
+	generatedToken := `sntrys_` + encodedJson + `_` + secrets.NewSecret(`[a-zA-Z0-9+/]{43}`)
+	tps = append(tps,
+		generatedToken,
+		"<token>"+generatedToken+"</token>",
+		"https://example.com?token="+generatedToken+"&other=stuff",
+	)
+
+	fps := []string{
+		secrets.NewSecret(`sntrys_[a-zA-Z0-9]{90}_[a-zA-Z0-9]{43}`),          // does not contain encoded json
+		`sntrys_` + encodedJson + `_` + secrets.NewSecret(`[a-zA-Z0-9]{42}`), // too short
+		`sntrys_` + encodedJson + `_` + secrets.NewSecret(`[a-zA-Z0-9]{44}`), // too long
+	}
+
+	return utils.Validate(r, tps, fps)
+}
+
+func SentryUserToken() *config.Rule {
+	// define rule
+	r := config.Rule{
+		RuleID:      "sentry-user-token",
+		Description: "Found a Sentry.io User Token, risking unauthorized access to error tracking services and sensitive application data.",
+		Regex:       utils.GenerateUniqueTokenRegex(`sntryu_[a-fA-F0-9]{64}`, false),
```

Comment:
Are we sure it's possible for these to contain uppercase? I've only found instances with lowercase.

Reply:
It would be a valid token, if I understand this rust parser correctly:
https://github.com/getsentry/sentry-cli/blob/c4b99eb4671ccb9146c71b18dc3a72a73d28c99f/src/utils/auth_token/user_auth_token.rs#L17C36-L17C55

But thinking about it, the prefix still has to be lower case, so the only situation this would be relevant is probably is someone converts the hex string to uppercase manually. Not sure if that is worth guarding against

Reply:
It looks like the tokens *should* be lowercase but it accepts uppercase as well and automatically converted them.

I suppose this is fine as-is.

Reply:
I have converted to rule to lower case only, all the leaked keys on github seem to be lower-case.